client/src/main/scala/drt/client/components/NegativeFeedbackComponent.scala

```diff
@@ -0,0 +1,137 @@
+package drt.client.components
+
+import drt.client.logger.{Logger, LoggerFactory}
+import drt.client.services.DrtApi
+import japgolly.scalajs.react.component.Scala.Component
+import japgolly.scalajs.react.vdom.html_<^.{<, VdomTagOf, ^, _}
+import japgolly.scalajs.react.{Callback, CtorType, ReactEventFromInput, Reusability, ScalaComponent}
+import org.scalajs.dom.html.{Anchor, Div}
+import uk.gov.homeoffice.drt.auth.LoggedInUser
+import upickle.default.{macroRW, write, ReadWriter => RW}
+
+
+object NegativeFeedbackComponent {
+  val log: Logger = LoggerFactory.getLogger(getClass.getName)
+
+  case class Props(url: String,
+                   loggedInUser: LoggedInUser)
+
+  case class State(url: String,
+                   feedbackUserEmail: String,
+                   whatUserDoing: String,
+                   whatWentWrong: String,
+                   whatToImprove: String,
+                   contactMe: Boolean,
+                   showDialogue: Boolean = false)
+
+  implicit val stateReuse: Reusability[State] = Reusability.derive[State]
+  implicit val propsReuse: Reusability[Props] = Reusability.by(p => (p.url))
+  implicit val rw: RW[State] = macroRW
+
+  val component: Component[Props, State, Unit, CtorType.Props] = ScalaComponent.builder[Props]("FeedbackComponent")
+    .initialStateFromProps(p => State(
+      url = p.url,
+      feedbackUserEmail = p.loggedInUser.email,
+      whatUserDoing = "",
+      whatWentWrong = "",
+      whatToImprove = "",
+      contactMe = false
+    ))
+    .renderPS((scope, props, state) => {
+
+      val showClass = if (state.showDialogue) "show" else "fade"
+
+      def setWhatUserDoing(whatUserDoing: String) = scope.modState(state => {
```

Comment:
These could be made a little more concise:
`scope.modState(_.copy(whatUserDoing = whatUserDoing))`